### PR TITLE
Integrate Mem0 for providing long-term memory for AI Agents

### PR DESCRIPTION
**Pull request by Dev-Khant**
## Checks

- [x] I've included any doc changes needed for https://microsoft.github.io/autogen/. See https://microsoft.github.io/autogen/docs/Contribute#documentation to build and test documentation locally.
- [x] I've added tests (if relevant) corresponding to the changes introduced in this PR.
- [x] I've made sure all auto checks have passed.

### CODE REVIEW

**Comment by Hk669** [comment]
@Dev-Khant can you please re-check the failing checks. thanks

**Comment by Dev-Khant** [comment]
> @Dev-Khant can you please re-check the failing checks. thanks

Hey I have made the formatting changes, all the pre-commit checks are passing locally.

**Review by Hk669 — APPROVED** [review]
looks good to me.

File: notebook/agentchat_with_memory.ipynb [hunk 1]
```diff
@@ -0,0 +1,345 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "A3guIYLG6PnX"
+   },
+   "source": [
+    "<a href=\"https://colab.research.google.com/drive/1NZEwC9w6V2S6hYmK7l2SQ9jhQrG1uKk8?usp=sharing\" target=\"_parent\"><img src=\"https://colab.research.google.com/assets/colab-badge.svg\" alt=\"Open In Colab\"/></a>"
```
Comment (sonichi) [inline notebook/agentchat_with_memory.ipynb]: This part can be removed and metadata can be added for rendering on the website. See https://github.com/autogen-ai/autogen/blob/main//notebook/agentchat_agentops.ipynb for an example.

**Review by sonichi — COMMENTED** [review]
Thanks for the contribution. The notebook contribution guide can be found at: https://github.com/autogen-ai/autogen/blob/main/notebook/contributing.md#how-to-get-a-notebook-displayed-on-the-website
The change to the existing two notebooks is not necessary.

File: notebook/agentchat_with_memory.ipynb (hunk 1 above)
Comment (Dev-Khant) [reply notebook/agentchat_with_memory.ipynb]: Hi @sonichi Thanks for sharing the example. I have made the necessary changes.

**Review by sonichi — APPROVED** [review]

File: notebook/agentchat_with_memory.ipynb (hunk 1 above)
Comment (sonichi) [reply notebook/agentchat_with_memory.ipynb]: Thanks. LGTM.

**Pull request merged**